pkg/collector/reconcile/opentelemetry.go

```diff
@@ -38,6 +39,17 @@ import (
 func Self(ctx context.Context, params manifests.Params) error {
 	changed := params.Instance
 
+	up := &collectorupgrade.VersionUpgrade{
+		Log:      params.Log,
+		Version:  version.Get(),
+		Client:   params.Client,
+		Recorder: params.Recorder,
+	}
+	changed, err := up.ManagedInstance(ctx, changed)
+	if err != nil {
+		return fmt.Errorf("failed to upgrade the OpenTelemetry CR: %w", err)
+	}
```

Comment:
on second look [here](https://github.com/iblancasa/opentelemetry-operator/blob/17e681d1df853be639c6351ad528677d8c33fba0/pkg/collector/upgrade/upgrade.go#L104-L108), it looks like this _may_ break users using custom images that don't adhere to semver. on failure to parse the instance, we would fail to call `updateScaleSubResourceStatus` which would mean any HPA would fail as well.

Reply:
From what I see, that problem will happen also with a regular upgrade, right?
If I'm rigth, I think it makes sense to fix that as part of another PR.

Reply:
yes and no, the current way we use this is as a [manager runnable task](https://github.com/open-telemetry/opentelemetry-operator/blob/98b2f3748afc43c79ce6336bb94925db6560c250/main.go#L303-L312) which wouldn't block setting the status.